pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -411,13 +411,14 @@ public class ModelInterpreter implements Serializable {
         if (when == null) {
             return true
         } else {
+            boolean ret = true
             for (int i = 0; i < when.conditions.size(); i++) {
                 DeclarativeStageConditional c = when.conditions.get(i)
                 if (!c.getScript(script).evaluate()) {
-                    return false
+                    ret = false
```

Comment:
This seems to be an unneeded change? Why would you need to iterate through them all?